### PR TITLE
Feature/ab#1017 dotnet tooling v2

### PR DESCRIPTION
**Pull request by Anthony-Mckale**
[AB#1017](https://dev.azure.com/riversafe/0b0abc1f-f46a-4d96-a470-e1c546e3cf29/_workitems/edit/1017)

added in combined SCA and SBOM dotnet-cyclonedx

features:
- gets vulnerabilities from dotnet command
- gets deprecated packages from dotnet command
- gets transitive packages from dotnet command
- gets detailed vulnerability sca data from osv

MINOR:
- offical -> official  spelling improvements
- fixed the trufflehog version to 3.0.4 due to latest being slightly unstable (dropped the --exclude flag...)
- made a standard INCLUDE_DEV flag and modified all tools to use this common flag
- added a is_transitive logic to SBOMs console so can print out top-level or "transitive" sub dependencies, PRINT_TRANSITIVE_PACKAGES flag in console.py
- removed .bom deprecated field from scan result
- ordered tools alphabetically
- refactored has_sbom_data / has_vulnerabilities logic into functions, in console/markdown.py
- maded find_files_by_name much safier (fully qualified regex matching with ^$)
- added truncate to sbom table (so newline and long text doesn't break the table)

### CODE REVIEW

File: eze/utils/language/dotnet.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                }
+            ),
+            extract_deprecated_packages(completed_process.stdout),
+        )
+    )
+
+
+def extract_vulnerable_packages(stdout: str) -> list:
+    """
+    extract deprecated packages from dotnet output
```
Comment (mgonzales-rs) [inline eze/utils/language/dotnet.py:87]: vulnerable *

Comment (Anthony-Mckale) [reply eze/utils/language/dotnet.py:87]: omdbljc... :( ugh

Comment (Anthony-Mckale) [reply eze/utils/language/dotnet.py:87]: fixed

File: eze/utils/language/dotnet.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        matches = deprecated_package_re.match(stdout_line)
+        if not matches:
+            continue
+        deprecated_packages.append(VulnerablePackage(matches.groups()))
+    return deprecated_packages
+
+
+async def get_vulnerable_packages(project_folder: str, dotnet_project_file: str) -> list:
+    """
+    use dotnet to get list of deprecated packages
```
[the remark this answers is not in the document]
Comment (Anthony-Mckale) [reply eze/utils/language/dotnet.py:103]: fixed

**Pull request merged**